torchdata/datapipes/iter/util/randomsplitter.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Tbh unsure. I made a new file since the use cases seem different enough and we might add more splitting functionality (such as other ones available in core).

We can potentially consolidate based on the categories in the [doc](https://pytorch.org/data/beta/torchdata.datapipes.iter.html)? Or deal with it when we move things from Core to here?